pilot/pkg/networking/core/v1alpha3/tracing.go

```diff
@@ -586,6 +593,101 @@ func proxyConfigSamplingValue(config *meshconfig.ProxyConfig) float64 {
 	return sampling
 }
 
+func configureTracingProvider(hcmTracing *hcm.HttpConnectionManager_Tracing, pushCtx *model.PushContext, proxyCfg *meshconfig.ProxyConfig, node *model.Proxy) {
+	if proxyCfg == nil || proxyCfg.Tracing == nil {
+		return
+	}
+	var serviceCluster string
+	if node.XdsNode != nil {
+		serviceCluster = node.XdsNode.Cluster
+	}
+
+	switch tracer := proxyCfg.Tracing.Tracer.(type) {
+	case *meshconfig.Tracing_Datadog_:
+		hostname, clusterName, err := getHostCluster(pushCtx, tracer.Datadog.Address)
+		if err != nil {
+			model.IncLookupClusterFailures("datadog")
+			return
+		}
+
+		cfg, _ := datadogConfig(serviceCluster, hostname, clusterName)
+		hcmTracing.Provider = &tracingcfg.Tracing_Http{
+			Name: envoyDatadog,
+			ConfigType: &tracingcfg.Tracing_Http_TypedConfig{
+				TypedConfig: cfg,
+			},
+		}
+	case *meshconfig.Tracing_Lightstep_:
+		hostname, clusterName, err := getHostCluster(pushCtx, tracer.Lightstep.Address)
+		if err != nil {
+			model.IncLookupClusterFailures("lightstep")
+			return
+		}
+
+		cfg, _ := otelLightStepConfig(clusterName, hostname, tracer.Lightstep.GetAccessToken())
+		hcmTracing.Provider = &tracingcfg.Tracing_Http{
+			Name: envoyOpenTelemetry,
+			ConfigType: &tracingcfg.Tracing_Http_TypedConfig{
+				TypedConfig: cfg,
+			},
+		}
+	case *meshconfig.Tracing_OpenCensusAgent_:
+	case *meshconfig.Tracing_Stackdriver_:
+		// stackdriver/opencensus always need bootstrap tracing?
+	case *meshconfig.Tracing_Zipkin_:
+		hostname, clusterName, err := getHostCluster(pushCtx, tracer.Zipkin.Address)
+		if err != nil {
+			log.Warnf("get host cluster failed: %v", err)
+			model.IncLookupClusterFailures("zipkin")
+			return
+		}
+		cfg, _ := zipkinConfig(hostname, clusterName, false)
+		hcmTracing.Provider = &tracingcfg.Tracing_Http{
+			Name: envoyZipkin,
+			ConfigType: &tracingcfg.Tracing_Http_TypedConfig{
+				TypedConfig: cfg,
+			},
+		}
+	default:
+		log.Warn("Tracing provider not configured")
+	}
+}
+
+func getHostCluster(push *model.PushContext, address string) (string, string, error) {
+	host, port, err := net.SplitHostPort(address)
+	if err != nil {
+		return "", "", fmt.Errorf("could not parse address %q: %v", address, err)
+	}
+	h := model.ResolveShortnameToFQDN(host, config.Meta{
+		Namespace: push.Mesh.RootNamespace,
```

Comment:
technically the old behavior was the proxy namespace, not RootNamespace...

This would be a really weird config though

Reply:
this something in MeshConfig, I think there are several situations:
1. fqdn 
2. servicename only
3. servicename.namepace

for (1) and (3), nothing to do;
for (2), should we use rootnamespace instead of proxy namespace?